### PR TITLE
Lane blobs

### PR DESCRIPTION
**Pull request by Darkwiiyou**
Blobs blobs and m ore blobs

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -49,6 +74,10 @@ def draw():
 
     for entity in entities:
         WINDOW.blit(entity.sprite.img, entity.pos, entity.sprite.rect)
+        pygame.font.init()
+        myfont = pygame.font.SysFont('Comic Sans MS', 30)
```
Comment (Jayx20) [inline main.py]: we should do these in init(), not every frame

File: main.py [hunk 2]
```diff
@@ -49,6 +74,10 @@ def draw():
...
+        textsurface = myfont.render("Blobs: " + str(money), False, (0, 0, 0))
+        WINDOW.blit(textsurface,(700,0))
```
Comment (Jayx20) [inline main.py]: do these outside of the for loop, currently it would render and blit the text for every entity we display

File: main.py [hunk 3]
```diff
@@ -12,35 +13,59 @@
...
+    global blob
+    global money
+    distanceX = test_entity.pos.x - blob.pos.x
+    distanceY = test_entity.pos.y - blob.pos.y
+    sprint = 1
+    prUpdate = False
+    # input stuff, move later to ma
     keys = pygame.key.get_pressed()
+    if keys[pygame.K_LSHIFT]:
+        sprint += 2
```
Comment (Jayx20) [inline main.py]: does sprint ever decrease? (is it supposed to?)

File: main.py [hunk 4]
```diff
@@ -12,35 +13,59 @@
...
-
-
 # runs every frame - game logic
 def update():
     global test_entity
-    print('test entity pos is '+str(test_entity.pos))
-
-    # input stuff, move later to make this less terrible / in the way
+    global blob
+    global money
```
Comment (Jayx20) [inline main.py]: just saying when you use `global` you can do like `global test_entity, blob, money`
(at least I think)

File: entity.py [hunk 5]
```diff
@@ -9,3 +9,4 @@ class Entity:
     def __init__(self, sprite_name: str, x: int, y: int):
         self.sprite = Sprite(sprite_name)
         self.pos = Vector2(x, y)
+        print("Sprite has been created with the name: " + sprite_name)
```
Comment (Jayx20) [inline entity.py]: this is good, later we should probably make some `if` check for if we want debug logs like you have with the position logging in main

**Review by Jayx20 — CHANGES_REQUESTED** [review]
Will merge after text stuff is fixed

File: main.py [hunk 6]
```diff
@@ -29,9 +29,6 @@ def init():
     entities.append(blob)
     test_entity = Entity("goodsprite.png", 5, 5)
     entities.append(test_entity)
-    # create joe and add him to the game like an epic boss
-    joe = Entity("goodsprite.png", 10, 10)
-    entities.append(joe)
```
Comment (Jayx20) [inline main.py:25]: thats fine I just added him for a test

File: main.py (hunk 1 above)
Comment (Darkwiiyou) [reply main.py]: Done

File: main.py (hunk 2 above)
Comment (Darkwiiyou) [reply main.py]: Done

File: main.py (hunk 3 above)
Comment (Darkwiiyou) [reply main.py]: I was going to make it togglable but decided it's fine as is

File: main.py (hunk 4 above)
Comment (Darkwiiyou) [reply main.py]: Will fix

File: entity.py (hunk 5 above)
Comment (Darkwiiyou) [reply entity.py]: true true, can we set it up to default as off for this specifically?

Comment (Jayx20) [reply entity.py]: Leave it for now I'll make an issue for it

**Comment by Darkwiiyou** [comment]
Note for self: Rewrite blob program to allow for more than 1 blob and also iteration through blob coordinates to allow for scaling.
Blob into list
Do normal location check but compare for each blob in for each loop

Eventually migrate to calculating closest blob while moving to help with workload

**Review by Jayx20 — APPROVED** [review]

**Pull request merged**